swapserverrpc/server.proto

```diff
@@ -733,3 +753,157 @@ message ServerWithdrawResponse {
     // The nonces that the server used to generate the sweepless sweep sigs.
     repeated bytes server_nonces = 2;
 }
+
+message ServerStaticAddressLoopInRequest {
+    // The client's public key for the htlc output.
+    bytes htlc_client_pub_key = 1;
+
+    // The hashed swap invoice preimage of the swap.
+    bytes swap_hash = 2;
+
+    // The deposit outpoints the client wishes to loop in. They implicitly state
+    // the swap amount.
+    repeated string deposit_outpoints = 3;
+
+    // The swap invoice that the client wants the server to pay.
+    string swap_invoice = 4;
+
+    // An optional last hop the client wants to receive the invoice payment
+    // from.
+    bytes last_hop = 5;
+
+    // The protocol version that the client adheres to.
+    StaticAddressProtocolVersion protocol_version = 6;
+
+    // The user agent string that identifies the software running on the user's
+    // side. This can be changed in the user's client software but it _SHOULD_
+    // conform to the following pattern:
+    //    Agent-Name/semver-version(/additional-info)
+    // Examples:
+    //    loopd/v0.10.0-beta/commit=3b635821
+    //    litd/v0.2.0-alpha/commit=326d754
+    //    loopd/v0.10.0-beta/commit=3b635823,initiator=easy-autoloop
+    string user_agent = 7;
+
+    // The swap payment timeout allows the user to specify an upper limit for
+    // the amount of time the server is allowed to take to fulfill the off-chain
+    // swap payment. If the timeout is reached the swap will be aborted on the
+    // server side and the client can retry the swap with different parameters.
+    uint32 payment_timeout_seconds = 8;
+}
+
+message ServerStaticAddressLoopInResponse {
+    // The server's public key for the htlc output.
+    bytes htlc_server_pub_key = 1;
+
+    // The cltv expiry height for the htlc. This is the height at which the
+    // htlc will expire and the client is free to claim the funds back.
+    int32 htlc_expiry = 2;
+
+    // The info the server used to generate the standard fee partial htlc tx
+    // sigs.
+    ServerHtlcSigningInfo standard_htlc_info = 3;
```

Comment:
these maps don't allow enum keys, just arbitrary <string,string> mappings, so we'd have to setup keys for this map in code rather than in the protobuf definition, which makes it a little less clear what's transported imo.